plugins/reporters/spdx/src/main/kotlin/SpdxDocumentReporter.kt

```diff
@@ -167,3 +168,23 @@ private fun SpdxExpression.getLicenseRefExceptions(result: MutableSet<String>) {
         else -> {}
     }
 }
+
+private fun ObjectNode.patchSpdx23To22(): ObjectNode {
+    put("spdxVersion", "SPDX-2.2")
+    val packages = get("packages") ?: return this
+
+    packages.forEach { pkg ->
+        val extRefs = pkg.get("externalRefs") ?: return@forEach
+
+        extRefs.forEach { ref ->
+            val refCat = ref.get("referenceCategory") ?: return@forEach
+
+            val refCatText = refCat.textValue()
+            if (refCatText != null && "-" in refCatText) {
+                (ref as ObjectNode).put("referenceCategory", refCatText.replace("-", "_"))
+            }
+        }
+    }
+
+    return this
```

Comment:
I hope my comment (on the outdated diff) was not mis-leading.

If we want to use this function (as-is) also for the new reporter (Spdx Java Library), then it should probably use KxS (and kaml, for the yaml case), and take a string as input.

Should we do this already now, or do you prefer to keep the function as-is and refactor it later again?